pkg/internal/codegen/parse/index.go

```diff
@@ -74,8 +74,15 @@ func (b *APIs) parseIndex() {
 			r.Resource = strings.ToLower(inflect.Pluralize(r.Kind))
 		}
 		// rt := parseResourceTag(b.getResourceTag(c))
-		// r.Resource = rt.Resource
-		// r.ShortName = rt.ShortName
+		rt, err := parseResourceAnnotation(c)
+		if err != nil {
+			log.Fatalf("failed to parse resource annotations, error: %v", err.Error())
+		}
+		if rt.Resource != "" {
+			r.Resource = rt.Resource
+		}
+		r.ShortName = rt.ShortName
```

Comment:
`If` condition for this code is handled in the crd.go file (https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/internal/codegen/parse/crd.go#L123)